src/desktop/apps/artwork/components/auction/view.coffee

```diff
@@ -62,13 +64,30 @@ module.exports = class ArtworkAuctionView extends Backbone.View
   acquire: (e) ->
     e.preventDefault()
 
-    $target = $(e.currentTarget)
-    $target.attr 'data-state', 'loading'
+    # Show the new buy now flow if you have the lab feature enabled
+    loggedInUser = CurrentUser.orNull()
+    if loggedInUser?.hasLabFeature('New Buy Now Flow')
+      serializer = new Serializer @$('.js-artwork-auction-buy-now-form')
+      data = serializer.data()
+
+      createOrder
+        artworkId: @data.artwork.get('_id')
+        editionSetId: data.edition_set_id
```

Comment:
I'm pretty unsure on this whole part. The form doesn't (as far as I'm aware) have the `edition_set_id` on it. I'm just not really sure where to get that info from. Also, not sure if I'm accessing `artwork` correctly above that.

Reply:
I think you're right-- it looks like we only ever pass the `artwork_id` for an auction work (and I seem to recall us not supporting being able to bid/buy multiple edition sets). So I think we can assume that we'll only have an artwork.

Looking below at the existing `acquire` method, it seems that we can get the `artworkId` from `AUCTION.artwork_id`. This gets defined [here](https://github.com/artsy/force/blob/master/src/desktop/apps/artwork/components/auction/bootstrap.coffee#L4) and set on `sd` [here](https://github.com/artsy/force/blob/master/src/desktop/apps/artwork/routes.coffee#L71) and [here](https://github.com/artsy/force/blob/master/src/desktop/apps/artwork/routes.coffee#L87).